### PR TITLE
Demonstration Notebook for IMDB-to-Hotels

### PR DESCRIPTION
**Pull request by riedgar-ms**
Adding a sample notebook which showcases the usage of AdaTest to convert a model pretrained on IMDB sentiment data to classify tweets about hotels. The finetuning is performed on the same machine running the notebook.

### CODE REVIEW

File: notebooks/IMDB to Hotel Sentiment.ipynb [hunk 1]
```diff
@@ -0,0 +1,725 @@
...
+   "id": "4d90d0d8",
+   "metadata": {},
+   "source": [
+    "## Using AdaTest\n",
+    "\n",
+    "AdaTest is a tool to help create training/test suites for language models. The basic workflow is:\n",
+    "\n",
+    "1. User provides some sample input\n",
+    "1. User flags whether the model output is correct or not\n",
+    "1. AdaTest uses a second language model to general more inputs from those already provided\n",
```
Comment (Harsha-Nori) [inline notebooks/IMDB to Hotel Sentiment.ipynb:122]: "general" -> "generate"?

File: notebooks/IMDB to Hotel Sentiment.ipynb [hunk 2]
```diff
@@ -0,0 +1,725 @@
...
+   "source": [
+    "generator = adatest.generators.OpenAI('curie', api_key=OPENAI_API_KEY)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "56095496",
+   "metadata": {},
+   "source": [
+    "Now we create the test tree. We will load a set of tests which we have alreadys started work on, to make the process faster:"
```
Comment (Harsha-Nori) [inline notebooks/IMDB to Hotel Sentiment.ipynb:181]: alreadys -> already

File: notebooks/IMDB to Hotel Sentiment.ipynb [hunk 3]
```diff
@@ -0,0 +1,725 @@
...
+    "    # Remove columns we don't need\n",
+    "    tmp4 = tmp3.drop(labels=['labeler', 'description', 'author', 'Unnamed: 0'], axis=1)\n",
+    "    \n",
+    "    # Rename columns\n",
+    "    tmp5 = tmp4.rename(mapper={'input': 'sentence', 'label': 'model_is_correct'}, axis=1)\n",
+    "    \n",
+    "    # Don't need to track original rows\n",
+    "    tmp6 = tmp5.reset_index(drop=True)\n",
+    "    \n",
+    "    return tmp6\n",
```
Comment (Harsha-Nori) [inline notebooks/IMDB to Hotel Sentiment.ipynb:246]: I don't think we should end up with an 'Unnamed: 0' in the columns when we read CSV -- maybe we need to set header or index to false on the CSV load?

File: notebooks/IMDB to Hotel Sentiment.ipynb [hunk 4]
```diff
@@ -0,0 +1,725 @@
...
+    "\n",
+    "display(test_data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "385dbdff",
+   "metadata": {},
+   "source": [
+    "We can also evaluate our chosen metric, and check that the accuracy score matches that we expect from the summary at the top level of the AdaTest widget:"
```
Comment (Harsha-Nori) [inline notebooks/IMDB to Hotel Sentiment.ipynb:342]: "that we expect" -> "what we expect"

File: notebooks/IMDB to Hotel Sentiment.ipynb [hunk 5]
```diff
@@ -0,0 +1,725 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "20d7fbf6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_df, test_df = train_test_split(test_data, stratify=test_data['topic'], test_size=0.3)"
+   ]
+  },
```
Comment (Harsha-Nori) [inline notebooks/IMDB to Hotel Sentiment.ipynb:419]: Let's merge these cells to keep things a bit more compact.

File: notebooks/imdb_hotel_conversion.csv [hunk 6]
```diff
@@ -0,0 +1,200 @@
+,topic,input,output,label,labeler,description,author,model score
```
Comment (Harsha-Nori) [inline notebooks/imdb_hotel_conversion.csv:1]: I think it's fine here for now, but let's keep an eye on creating a subfolder for test trees inside the notebooks directory in the future.

**Review by Harsha-Nori — COMMENTED** [review]
Looking good! Suggested a few small tweaks and typo fixes. In general I think it'd be nice if we can push for more a bit more compactness. Really like the functionality!

File: notebooks/IMDB to Hotel Sentiment.ipynb (hunk 1 above)
Comment (riedgar-ms) [reply notebooks/IMDB to Hotel Sentiment.ipynb:122]: Thanks

File: notebooks/IMDB to Hotel Sentiment.ipynb (hunk 2 above)
Comment (riedgar-ms) [reply notebooks/IMDB to Hotel Sentiment.ipynb:181]: Indeed

File: notebooks/IMDB to Hotel Sentiment.ipynb (hunk 3 above)
Comment (riedgar-ms) [reply notebooks/IMDB to Hotel Sentiment.ipynb:246]: It's not that. It's the CSV itself:
```
,topic,input,output,label,labeler,description,author,model score
00f2dfd40977438f9942793768959e8a,suggestion,,,topic_marker,imputed,,,
02076434c5fc4a389f127e542c479c64,/Rooms,Sheets didn't seem clean,neg,pass,anonymous,,anonymous,-0.9646334648132324
```
There is no name for the column of GUIDs. That's what's being dropped.

I'll have this down as a "Won't fix" for this notebook. Perhaps AdaTest should put a name on that column, though.

File: notebooks/IMDB to Hotel Sentiment.ipynb (hunk 4 above)
Comment (riedgar-ms) [reply notebooks/IMDB to Hotel Sentiment.ipynb:342]: Agreed

File: notebooks/IMDB to Hotel Sentiment.ipynb (hunk 5 above)
Comment (riedgar-ms) [reply notebooks/IMDB to Hotel Sentiment.ipynb:419]: OK

File: notebooks/imdb_hotel_conversion.csv (hunk 6 above)
Comment (riedgar-ms) [reply notebooks/imdb_hotel_conversion.csv:1]: In the future I will have plenty of suggestions for handling notebooks :-)

**Review by Harsha-Nori — APPROVED** [review]
Looks good -- thanks for pulling this together!

**Pull request merged**